pkg/apis/core/validation/validation.go

```diff
@@ -5530,7 +5530,6 @@ func ValidateResourceQuantityValue(resource string, value resource.Quantity, fld
 // newResourceQuota is updated with fields that cannot be changed.
 func ValidateResourceQuotaUpdate(newResourceQuota, oldResourceQuota *core.ResourceQuota) field.ErrorList {
```

Comment:
I would actually go the other direction and drop the `resourcequotaStrategy#ValidateUpdate` strategy call to `ValidateResourceQuota`:

* `ValidateObjectMetaUpdate` is sufficient on update, we don't need to call `ValidateObjectMeta` again
* `ValidateResourceQuotaSpec` is already being called from ValidateResourceQuotaUpdate and makes sense
* `newResourceQuota.Status = oldResourceQuota.Status` is already done in `resourcequotaStrategy#PrepareForUpdate`, so the validation method shouldn't do it again (mutating an input in validation is strange anyway)
* Since updates can't change status, it doesn't make sense to validate them in `ValidateResourceQuotaUpdate` either

Reply:
Yes. Agree.

My first try to fix is to add check after resourcequotaStrategy#ValidateUpdate
```
// ValidateUpdate is the default update validation for an end user.
func (resourcequotaStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
	errorList := validation.ValidateResourceQuota(obj.(*api.ResourceQuota))
+	if len(errorList) != 0 {
+		return errorList
+	}
	return append(errorList, validation.ValidateResourceQuotaUpdate(obj.(*api.ResourceQuota), old.(*api.ResourceQuota))...)
}
```

Your suggestion would be more direct.

Updated the PR. Thanks for your comments @liggitt.